### DISCUSSION TITLE
v3.8.0_z: Version 12.72 and new features

### DISCUSSION THREAD

**Discussion by Zbizu** [Zbizu/rme#2] [Announcements]
## New features
- client 12.71/12.72
- new OTB (version 3.59/client 12.72, developed with approval of otland staff so this will be official one, backwards compatible with EPuncker's OTB)
- colourable mounts
- podium support (can be enabled two ways: item group `14` (`0x0E`) in **items.otb** or `<attribute key="type" value="podium" />` in **items.xml**)
![obraz](https://user-images.githubusercontent.com/5205079/141848118-d3f45ab1-da4b-49bb-88cf-f90887c399f5.png)
- remove all invalid item ids from the map (maps with invalid items visible as red squares fail to load on the servers, this feature removes all of them)
![obraz](https://user-images.githubusercontent.com/5205079/142282324-5c3f0e38-afec-42ae-afb8-778d4cd57d2e.png)

## Technical changes
- invisible wall 1548 now displayed as cyan square for better visibility:
![obraz](https://user-images.githubusercontent.com/5205079/141848563-f7a81ed1-6b3c-4a4c-ae36-5a8a4482dd2e.png)
- slight cosmetic changes to properties dialog buttons
- merged two functions responsible for drawing items into one (only one parameter was different so there was no need to have two large, almost identical functions)
- features such as "find by id", "remove by id" and "replace one item with another" now allow forcing server ID as another way to combat red squares
![obraz](https://user-images.githubusercontent.com/5205079/142282370-939e65ca-8019-4125-97a7-9a96cd1e6db6.png)

## Possible compatibility with
- https://github.com/Zbizu/forgottenserver (full compatibility)
- https://github.com/opentibiabr/canary (requires https://github.com/opentibiabr/canary/pull/143, https://github.com/opentibiabr/canary/pull/161 << READ DESCRIPTION IF YOU WANT TO USE IT ON CANARY!)

## Podium format specification
legend:
1 byte = u8
2 bytes = u16

structure:
- 1 byte header: OTBM_ATTR_PODIUMOUTFIT (40 dec, 0x28 hex)
- 1 byte flags (showPlatform, showOutfit, showMount)
- 1 byte direction
- 2 bytes looktype
- 4x 1 byte outfit colors
- 1 byte addons
- 2 bytes mount looktype
- 4x 1 byte mount colors

## Notes
- As this is a new feature, any maps containing podiums may fail to load in outdated server engines unless they get patched with podium parsing (see https://github.com/opentibiabr/canary/pull/161/files for `propstream.read` example).

<hr /><em>This discussion was created from the release <a href='https://github.com/Zbizu/rme/releases/tag/v3.8.0_z'>v3.8.0_z: Version 12.72 and new features</a>.</em>

**Comment by elewental** [comment]
Great release, Thank You!
